core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +77,13 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
```

Comment:
I think we need to support backcompat here. You can look at version.created in indexSettings, and then only accept the old setting name if before 2.0 final. It would be nicer if we forced the upgrade once, when loading the index, but I don't think that is currently possible without a bunch more work.